src/realm/query_engine.cpp

```diff
@@ -175,6 +175,228 @@ size_t ParentNode::aggregate_local(QueryStateBase* st, size_t start, size_t end,
     }
 }
 
+void StringNodeEqual::deallocate() noexcept
+{
+    // Must be called after each query execution to free temporary resources used by the execution. Run in
+    // destructor, but also in Init because a user could define a query once and execute it multiple times.
```

Comment:
if it must be run *after* each execution, then why in init()?! After thinking about it I know what you mean, but it's stated confusingly